plugins/gradle/src/org/jetbrains/plugins/gradle/internal/daemon/DaemonAction.java

```diff
@@ -28,7 +32,22 @@ protected ServiceRegistry getDaemonServices(DaemonClientFactory daemonClientFact
     if (myServiceDirectoryPath != null && !myServiceDirectoryPath.isEmpty()) {
       layout.setGradleUserHomeDir(new File(myServiceDirectoryPath));
     }
-    DaemonParameters daemonParameters = new DaemonParameters(layout);
+    DaemonParameters daemonParameters = null;
+    boolean isGradle5Dot3OrNewer = GradleVersion.current().compareTo(GradleVersion.version("5.3")) >= 0;
+    if (!isGradle5Dot3OrNewer) {
+      daemonParameters = new DaemonParameters(layout);
+    }
+    else {
+      try {
+        //noinspection JavaReflectionMemberAccess
+        daemonParameters = DaemonParameters.class.getConstructor(BuildLayoutParameters.class, FileCollectionFactory.class)
+          .newInstance(layout, new DefaultFileCollectionFactory());
+      }
+      catch (InstantiationException | IllegalAccessException | InvocationTargetException | NoSuchMethodException e) {
+        Logging.getLogger(this.getClass())
```

Comment:
@vladsoroka I am not sure how to handle the exceptions here, do we want to re-throw it? Or make this method return nullable ServiceRegistry?

Reply:
Let's re-throw exceptions, it will be handled at https://github.com/JetBrains/intellij-community/blob/191/plugins/gradle/src/org/jetbrains/plugins/gradle/internal/daemon/GradleDaemonServices.groovy#L162-L170

Reply:
Thanks. Done.